lib/pipekit/field_repository.rb

```diff
@@ -0,0 +1,25 @@
+module Pipekit
+  module FieldRepository
+      def get_by_key(key)
+        key = Config.field(parent_resource, key)
+        search_fields("key", key)
+      end
+
+      def get_by_name(name)
+        search_fields("name", name)
+      end
+
+      private
+
+      def search_fields(field_element, value)
+        result = request.get.select { |element| element[field_element] == value }
```

Comment:
Feels a little bit strange to depend on `request`, an attribute set by the parent class. It's not obvious that to reuse this module we need to ensure a `#request` method is present in the parent class. Is that a common pattern for composition with modules?